Formula/y/yass-cli.rb

```diff
@@ -0,0 +1,70 @@
+class YassCli < Formula
+  desc "Lightweight and efficient, socks/http forward proxy"
+  homepage "https://letshack.info"
+  url "https://github.com/Chilledheart/yass/releases/download/1.13.1/yass-1.13.1.tar.bz2"
+  sha256 "26015d32a6bdc5bf2670b36c8f5fb68f4a324d577e027bc5eef18ae977e00c9f"
+  license "GPL-2.0-only"
+  head "https://github.com/Chilledheart/yass.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "go" => :build
+  depends_on "llvm" => :build if DevelopmentTools.clang_build_version < 1500
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "c-ares"
+  depends_on "libnghttp2"
+  depends_on "mbedtls"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "zlib"
+  end
+
+  def install
+    ENV.llvm_clang if OS.mac? && (DevelopmentTools.clang_build_version < 1500)
+
+    args = %w[
+      -DBUILD_SHARED_LIBS=off
+      -DUSE_BUILTIN_CA_BUNDLE_CRT=off
+      -DCLI=on
+      -DSERVER=off
+      -DGUI=off
+      -DBUILD_TESTS=off
+      -DUSE_SYSTEM_MBEDTLS=on
+      -DUSE_ZLIB=on
+      -DUSE_SYSTEM_ZLIB=on
+      -DUSE_CARES=on
+      -DUSE_SYSTEM_CARES=on
+      -DUSE_SYSTEM_NGHTTP2=on
+    ]
```

Comment:
```suggestion
      -DUSE_SYSTEM_NGHTTP2=on
      -DUSE_TCMALLOC=off
      -DUSE_MIMALLOC=off
      -DUSE_SQLITE=off
      -DUSE_SYSTEM_JSON=on
    ]
```
Let's avoid using vendored stuff unintentionally. You'll probably also need
```ruby
depends_on "nlohmann-json" => :build
```

Reply:
tcmalloc and mimalloc are not available on macos (at least no working with yass).

I will add `-DUSE_SYSTEM_JSON=on`

Reply:
<img width="557" alt="截屏2024-09-08 20 00 11" src="https://github.com/user-attachments/assets/2e641e7e-d448-47d4-9e9b-8a288421f78d">

Reply:
Yes, I saw that, but I don't want to rely on those `cmake_dependent_option` declarations not changing.

Reply:
`cmake_dependent_option` means the if prerequirement condition is not met, there is no such cmake option. It makes no effect even if you pass an ON to such option.

Reply:
We can add it for linux through.